packages/contracts/contracts/core/DAO.sol

```diff
@@ -10,96 +10,116 @@ import "@openzeppelin/contracts/utils/Address.sol";
     using Address for address;
 
-    // Roles
-    bytes32 public constant UPGRADE_ROLE = keccak256("UPGRADE_ROLE");
-    bytes32 public constant DAO_CONFIG_ROLE = keccak256("DAO_CONFIG_ROLE");
-    bytes32 public constant EXEC_ROLE = keccak256("EXEC_ROLE");
-    bytes32 public constant WITHDRAW_ROLE = keccak256("WITHDRAW_ROLE");
-    bytes32 public constant SET_SIGNATURE_VALIDATOR_ROLE =
-        keccak256("SET_SIGNATURE_VALIDATOR_ROLE");
-    bytes32 public constant MODIFY_TRUSTED_FORWARDER = keccak256("MODIFY_TRUSTED_FORWARDER");
+    /// @notice The ID of the permission required for the `_authorizeUpgrade` function.
+    bytes32 public constant UPGRADE_PERMISSION_ID = keccak256("UPGRADE_PERMISSION_ID");
 
+    /// @notice The ID of the permission required for the `setMetadata` function.
+    bytes32 public constant SET_METADATA_PERMISSION_ID = keccak256("SET_METADATA_PERMISSION_ID");
+
+    /// @notice The ID of the permission required for the `execute` function.
+    bytes32 public constant EXECUTE_PERMISSION_ID = keccak256("EXECUTE_PERMISSION_ID");
+
+    /// @notice The ID of the permission required for the `withdraw` function.
+    bytes32 public constant WITHDRAW_PERMISSION_ID = keccak256("WITHDRAW_PERMISSION_ID");
+
+    /// @notice The ID of the permission required for the `setSignatureValidator` function.
+    bytes32 public constant SET_SIGNATURE_VALIDATOR_PERMISSION_ID =
+        keccak256("SET_SIGNATURE_VALIDATOR_PERMISSION_ID");
+
+    /// @notice The ID of the permission required for the `setTrustedForwarder` function.
+    bytes32 public constant SET_TRUSTED_FORWARDER_PERMISSION_ID =
+        keccak256("SET_TRUSTED_FORWARDER_PERMISSION_ID");
+
+    /// @notice The [ERC-1271](https://eips.ethereum.org/EIPS/eip-1271) signature validator contract.
     ERC1271 signatureValidator;
 
-    address private _trustedForwarder;
+    /// @notice The address of the trusted forwarder verifying meta transactions.
+    address private trustedForwarder;
 
-    /// @notice Thrown if action execution has failed
+    /// @notice Thrown if action execution has failed.
     error ActionFailed();
 
-    /// @notice Thrown if the deposit or withdraw amount is zero
+    /// @notice Thrown if the deposit or withdraw amount is zero.
     error ZeroAmount();
 
-    /// @notice Thrown if the expected and actually deposited ETH amount mismatch
-    /// @param expected Expected ETH amount
-    /// @param actual Actual ETH amount
-    error ETHDepositAmountMismatch(uint256 expected, uint256 actual);
-
-    /// @notice Thrown if an ETH withdraw fails
-    error ETHWithdrawFailed();
-
-    /// @dev Used for UUPS upgradability pattern
-    /// @param _metadata IPFS hash that points to all the metadata (logo, description, tags, etc.) of a DAO
+    /// @notice Thrown if the expected and actually deposited native token amount mismatch.
+    /// @param expected The expected native token amount.
+    /// @param actual The actual native token amount.
```

Comment:
Done.
@juliettech13: Do you know that you can make replacement suggestions in GH? This would make applying your comments easier. 
See here: https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review